crates/worker/src/partition_processor_manager/spawn_processor_task.rs

```diff
@@ -150,6 +150,8 @@ impl SpawnPartitionProcessorTask {
                         Err(e) => Err(ProcessorError::from(e)),
                     }?;
 
+                    partition_store.verify_deprecations(partition.key_range);
```

Comment:
I'm not sure we should do operations like these on startup. If this is something that can happen lazily (potentially after a partition becomes leader) then we should do that.

Reply:
So you would move this in Leadership state?